src/lib.rs

```diff
@@ -237,7 +250,34 @@ pub use crate::error::Error;
 // The function MUST NOT ever write uninitialized bytes into `dest`,
 // regardless of what value it returns.
 cfg_if! {
-    if #[cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))] {
+    if #[cfg(getrandom_backend = "custom")] {
+        #[path = "custom.rs"] mod imp;
+    } else if #[cfg(getrandom_backend = "linux_getrandom")] {
+        #[cfg(not(any(target_os = "android", target_os = "linux")))]
+        compile_error!("`linux_getrandom` backend can be enabled only for Linux/Android targets!");
+        mod util_libc;
+        #[path = "linux_android.rs"] mod imp;
+    } else if #[cfg(getrandom_backend = "rdrand")] {
+        #[cfg(not(any(target_arch = "x86_64", target_arch = "x86")))]
+        compile_error!("`rdrand` backend can be enabled only for x86 and x86-64 targets!");
+        #[path = "rdrand.rs"] mod imp;
+    } else if #[cfg(getrandom_backend = "wasm_js")] {
+        #[cfg(not(all(
+            any(target_arch = "wasm32", target_arch = "wasm64"),
+            target_os = "unknown",
+        )))]
+        compile_error!("`wasm_js` backend can be enabled only on OS-less WASM targets!");
+        #[path = "js.rs"] mod imp;
+    } else if #[cfg(getrandom_backend = "esp_idf")] {
+        #[cfg(not(target_os = "espidf"))]
+        compile_error!("`esp_idf` backend can be enabled only for ESP-IDF targets!");
```

Comment:
ESP-IDF support is now optional because of the concerns raised in https://github.com/rust-random/getrandom/issues/397.